### PR TITLE
User entity

### PR DESCRIPTION
**Pull request by SaboniAmine**
User management router & refacto

### CODE REVIEW

File: carbonserver/tox.ini [hunk 1]
```diff
@@ -8,7 +8,7 @@ deps =
 
 commands =
     pip install -e .
-    python -m pytest
+    python -m pytest -vv
```
Comment (benoit-cty) [inline carbonserver/tox.ini:11]: Are you sure you want to commit this ?

File: carbonserver/carbonserver/api/infra/repositories/repository_users.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        user = [user for user in self.users if user.user_id == user_id][0]
+        return user
+
+    def list_users(self):
+        user_ids = [user.user_id for user in self.users]
+        return user_ids
+
+
+def _hash_password(password: str) -> SecretStr:
+    salt = _random_generator()  # Extract random generation to make it easier to mock
```
Comment (benoit-cty) [inline carbonserver/carbonserver/api/infra/repositories/repository_users.py:43]: How can you check the password of the user with a random salt when he will log in ?
We have to store the salt in the database, with the user's information, isn't it ?

**Review by benoit-cty — COMMENTED** [review]
Thanks Amine. I put my questions in code.

File: carbonserver/carbonserver/api/infra/repositories/repository_users.py (hunk 2 above)
Comment (SaboniAmine) [reply carbonserver/carbonserver/api/infra/repositories/repository_users.py]: Indeed, the salt should be stored too. However this implementation would require a lot of User internal methods to check the password validity (request salt + hashed key from db, hash the user input and compare results) or to update it.
I found another implementation that seems interesting : https://variable-scope.com/posts/storing-and-verifying-passwords-with-sqlalchemy. We could first use secret passwords for the first version, and try to implement this authentication logic after. Wdyt ?

File: carbonserver/tox.ini (hunk 1 above)
Comment (SaboniAmine) [reply carbonserver/tox.ini:11]: I guess it will be useful when we will add those tests to the CI.

**Pull request merged**